sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -710,7 +710,7 @@ def get_datapath_init_duration(self):
         if self.is_flat_memory():
             return 0
         duration = self.xcvr_eeprom.read(consts.DP_PATH_INIT_DURATION)
-        return float(duration) if duration is not None else 0
+        return float(duration) if duration is not(None or 0) else 0
```

Comment:
For supported values this api returns non-zero value and for unsupported including error it returns 0

Reply:
Yes, with your code change in codes/public/cmis.py, we would set duration as 0 for unsupported (reserved) value. The existing if check (without your code changes in api/public/cmis.py) will return 0.0 and hence, no explicit handling is required in cmis.py.

In addition, the code change in PR will likely throw out an error if duration is None since we would try to return float(None) which is an invalid statement.